pyproject_metadata/__init__.py

```diff
@@ -229,6 +240,10 @@ def from_pyproject(
             msg = 'The description must be a single line'
             raise ConfigurationError(msg)
 
+        if metadata_version and metadata_version not in KNOWN_METADATA_VERSIONS:
```

Comment:
The first condition in the `if` is redundant

Reply:
It's not, this is an error branch and `None` is a legal value.

Reply:
Oh, you're right. I misread the code.

Reply:
I'm trying to keep the original behavior, though I'm not that fond of it - setting a value as dynamic (PEP 621) doesn't mean it's dynamic SDist -> wheel. Lots of examples, like readme generation, are still static.